cmd/pj-rehearse/main.go

```diff
@@ -161,14 +164,42 @@ func main() {
 		}
 	}
 
+	changedTemplates := make(map[string]*templateapi.Template)
 	// We can only detect changes if we managed to load both CI template versions
 	if masterConfig.Templates != nil && prConfig.Templates != nil {
-		changedTemplates := diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
+		changedTemplates = diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
 		for name := range changedTemplates {
 			logger.WithField("template-name", name).Info("Changed template")
 		}
 	}
 
+	var cmClient corev1.ConfigMapInterface
+	if !o.dryRun {
+		cmClient, err = rehearse.NewCMClient(clusterConfig, masterConfig.Prow.ProwJobNamespace, o.dryRun)
+		if err != nil {
+			logger.WithError(err).Error("could not create a configMap client")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+
+		cmController := config.NewTemplateCMController(cmClient, prNumber, logger, changedTemplates)
+		if errors := cmController.CreateCMTemplates(); len(errors) > 0 {
+			for _, err := range errors {
+				logger.WithError(err).Error("couldn't create template configMap")
+			}
+			gracefulExit(o.noFail, "")
+
+		}
+
+		defer func() {
```

Comment:
I don't think this will work properly - IIRC `defer` does not run on `os.Exit`, right? And we use that a lot for failure modes...

Reply:
I think we should cleanup the `configMaps` inside the `gracefulExit`.

Reply:
We should use https://golang.org/pkg/runtime/#Goexit and defer `os.Exit`

Reply:
We will do all the necessary cleanups in a `gracefullExit()` function. This can be done in a follow-up. For now, just deferring is ok.

Reply:
I **really** think we should do the proper cleanup in this PR, not a followup.